File: plugins/openclaw-bridge/openclaw-extension/src/model-catalog.ts
/** Report configured choices only. Do not claim a hard-coded model is installed. */
export function configuredModelCatalog(config: any, observedAt = new Date().toISOString()) {
 const defaults = config?.agents?.defaults ?? {};
 const configured = defaults.model;
 const primary = typeof configured === "string" ? configured : configured?.primary;
 const safe = (value: unknown): value is string => typeof value === "string" && value.length <= 200 && /^[A-Za-z0-9._:/-]+$/.test(value);
 const models = [...new Set([primary, ...(Array.isArray(configured?.fallbacks) ? configured.fallbacks : []), ...Object.keys(defaults.models ?? {})].filter(safe))] as string[];
 if (!safe(primary) || !models.length) return null;
 return { runtimeType: "openclaw", defaultModel: primary, models, source: "openclaw-configured-models", observedAt,
  entries: models.map(id => ({ id, model: id.startsWith("openrouter/") ? id.slice("openrouter/".length) : id,
   provider: id.startsWith("openrouter/") ? "openrouter" : id.split("/")[0], label: id, availability: "unknown" })) };
}

File: plugins/hermes-agent-bridge/src/model_catalog.py
"""Publish only the model selected in this Hermes installation."""
import re


def configured_model_catalog(config, observed_at):
    selected = config.get('model') if isinstance(config, dict) else None
    model = selected.get('default') if isinstance(selected, dict) else selected
    provider = selected.get('provider') if isinstance(selected, dict) else None
    if not isinstance(model, str) or not re.fullmatch(r'[A-Za-z0-9._:/-]{1,200}', model):
        return None
    entry = {'id': model, 'model': model, 'label': model, 'availability': 'unknown'}
    if isinstance(provider, str) and re.fullmatch(r'[A-Za-z0-9._-]{1,100}', provider):
        entry['provider'] = provider
    return {'runtimeType': 'hermes', 'defaultModel': model, 'models': [model],
            'entries': [entry], 'source': 'hermes-configured-model', 'observedAt': observed_at}
